### PR TITLE
[flink]CompactAction support catalog config

### PR DESCRIPTION
**Pull request by daixiaoyu**
[flink]

### Purpose

CompactAction support filesystem config

### Tests

*(List UT and IT cases to verify this change)*

### API and Format 

*(Does this change affect API or storage format)*

### Documentation
docs/content/maintenance/write-performance.md

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java [hunk 1]
```diff
@@ -74,6 +77,12 @@ public CompactAction withPartitions(List<Map<String, String>> partitions) {
         return this;
     }
 
+    static Options getOptions(MultipleParameterTool params) {
+        Options options = new Options();
+        params.toMap().forEach((k, v) -> options.set(k, v));
+        return options;
+    }
+
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java:85]: How about using `catalog-conf k=v`?
Like:
https://paimon.apache.org/docs/master/how-to/cdc-ingestion/#synchronizing-tables

Comment (daixiaoyu) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java:85]: That's a good idea, I'm going to modify it in this format

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java [hunk 2]
```diff
@@ -74,6 +78,13 @@ public CompactAction withPartitions(List<Map<String, String>> partitions) {
         return this;
     }
 
+    static Options getCatalogOptions(MultipleParameterTool params) {
+        Map<String, String> catalogConfig = getConfigMap(params, "catalog-conf");
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java:82]: `getConfigMap` may return null, so we need to check nullability.
Can we do it like this in `create`:
```
Tuple3<String, String, String> tablePath = getTablePath(params); 
if (tablePath == null) {
    return Optional.empty();
}
```
And then this method is not necessary.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java [hunk 3]
```diff
@@ -74,6 +78,13 @@ public CompactAction withPartitions(List<Map<String, String>> partitions) {
...
+        Options options = new Options();
+        catalogConfig.forEach((k, v) -> options.set(k, v));
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java:84]: `catalogConfig.forEach(options::set)`

File: docs/content/maintenance/write-performance.md [hunk 4]
```diff
@@ -165,8 +165,10 @@ Run the following command to submit a compaction job for the table.
     /path/to/paimon-flink-**-{{< version >}}.jar \
     compact \
     --warehouse <warehouse-path> \
-    --database <database-name> \
-    --table <table-name>
+    --database <database-name> \ 
+    --table <table-name> \
+    [--partition <partition-name>] \
+    [--catalog-conf <paimon-catalog-conf> [--catalog-conf <paimon-catalog-conf> ...]] \
```
Comment (yuzelin) [inline docs/content/maintenance/write-performance.md:171]: Add description for catalog-conf, like:
'--catalog-conf' is the configuration for Paimon catalog. Each configuration should be specified in the format 'key=value'. See [here]({{< ref "maintenance/configurations" >}}) for a complete list of catalog configurations.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ActionBase.java [hunk 5]
```diff
@@ -79,11 +79,9 @@ public abstract class ActionBase implements Action {
     }
 
     ActionBase(String warehouse, String databaseName, String tableName, Options options) {
+        options.set(CatalogOptions.WAREHOUSE, warehouse);
```
Comment (yuzelin) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ActionBase.java:82]: We need specify the options is for catalog, and table options can be handled in action implementation.

**Review by yuzelin — COMMENTED** [review]
Thanks, I left some comments.

**Review by yuzelin — APPROVED** [review]
LGTM. Only need a minor improve.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java [hunk 6]
```diff
@@ -74,6 +79,13 @@ public CompactAction withPartitions(List<Map<String, String>> partitions) {
         return this;
     }
 
+    static Options getCatalogOptions(MultipleParameterTool params) {
+        Optional<Map<String, String>> catalogConfigOption = getConfigMap(params, "catalog-conf");
+        Options options = new Options();
+        catalogConfigOption.ifPresent(catalogConfigMap -> catalogConfigMap.forEach(options::set));
+        return options;
```
[the remark this answers is not in the document]
Comment (daixiaoyu) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java]: Have been modified

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ActionBase.java [hunk 7]
```diff
@@ -79,11 +79,9 @@ public abstract class ActionBase implements Action {
     }
 
     ActionBase(String warehouse, String databaseName, String tableName, Options options) {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/ActionBase.java:81]: options -> catalogOptions

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java [hunk 8]
```diff
@@ -52,14 +54,18 @@ public class CompactAction extends ActionBase {
     private final CompactorSinkBuilder sinkBuilder;
 
     CompactAction(String warehouse, String database, String tableName) {
-        super(warehouse, database, tableName, new Options().set(CoreOptions.WRITE_ONLY, false));
+        this(warehouse, database, tableName, new Options());
+    }
+
+    CompactAction(String warehouse, String database, String tableName, Options options) {
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java:60]: options -> catalogOptions

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java [hunk 9]
```diff
@@ -52,14 +54,18 @@ public class CompactAction extends ActionBase {
...
+        super(warehouse, database, tableName, options);
         if (!(table instanceof FileStoreTable)) {
             throw new UnsupportedOperationException(
                     String.format(
                             "Only FileStoreTable supports compact action. The table type is '%s'.",
                             table.getClass().getName()));
         }
-
+        table = table.copy(new Options().set(CoreOptions.WRITE_ONLY, false).toMap());
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/CompactAction.java:68]: `Collections.singletonMap()`?

**Comment by JingsongLi** [comment]
Thanks @daixiaoyu and @yuzelin , we still have some small areas for improvement.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**